To: vim_dev@googlegroups.com
Subject: Patch 7.3.335
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.335
Problem:    When 'imdisable' is reset from an autocommand in Insert mode it
	    doesn't take effect.
Solution:   Call im_set_active() in Insert mode. (Taro Muraoka)
Files:	    src/option.c


*** ../vim-7.3.334/src/option.c	2011-09-21 13:40:13.000000000 +0200
--- src/option.c	2011-10-12 14:09:46.000000000 +0200
***************
*** 7806,7811 ****
--- 7806,7815 ----
  	/* Only de-activate it here, it will be enabled when changing mode. */
  	if (p_imdisable)
  	    im_set_active(FALSE);
+ 	else if (State & INSERT)
+ 	    /* When the option is set from an autocommand, it may need to take
+ 	     * effect right away. */
+ 	    im_set_active(curbuf->b_p_iminsert == B_IMODE_IM);
      }
  #endif
  
*** ../vim-7.3.334/src/version.c	2011-10-12 14:11:43.000000000 +0200
--- src/version.c	2011-10-12 16:56:30.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     335,
  /**/

-- 
Time is money.  Especially if you make clocks.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
